### PR TITLE
[Merged by Bors] - feat: (sSup, Iic) and (Ici, sInf) are Galois connections

### PR DESCRIPTION
**Pull request by ADedecker**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Set/Lattice.lean [hunk 1]
```diff
@@ -1159,6 +1159,17 @@ theorem sInter_eq_univ {S : Set (Set α)} : ⋂₀ S = univ ↔ ∀ s ∈ S, s =
...
 #align set.sInter_eq_univ Set.sInter_eq_univ
 
+theorem subset_powerset_iff {s : Set (Set α)} {t : Set α} : ⋃₀ s ⊆ t ↔ s ⊆ 𝒫 t :=
+  sUnion_subset_iff
+
+theorem sUnion_powerset_gc :
+    GaloisConnection (⋃₀ · : Set (Set α) → Set α) (𝒫 · : Set α → Set (Set α)) :=
+  gc_sSup_Iic
+
+theorem sUnion_powerset_gi :
```
Comment (jcommelin) [inline Mathlib/Data/Set/Lattice.lean:1169]:
```suggestion
def sUnion_powerset_gi :
```

File: Mathlib/Order/GaloisConnection.lean [hunk 2]
```diff
@@ -937,6 +945,14 @@ end lift
 
 end GaloisCoinsertion
 
+theorem gi_sSup_Iic [CompleteSemilatticeSup α] :
+    GaloisInsertion (sSup : Set α → α) (Iic : α → Set α) :=
+  gc_sSup_Iic.toGaloisInsertion fun _ ↦ le_sSup le_rfl
+
+theorem gci_Ici_sInf [CompleteSemilatticeInf α] :
```
Comment (jcommelin) [inline Mathlib/Order/GaloisConnection.lean:952]:
```suggestion
def gci_Ici_sInf [CompleteSemilatticeInf α] :
```

**Review by jcommelin — APPROVED** [review]
LGTM, but needs linting

bors d+

File: Mathlib/Data/Set/Lattice.lean [hunk 3]
```diff
@@ -1159,6 +1159,17 @@ theorem sInter_eq_univ {S : Set (Set α)} : ⋂₀ S = univ ↔ ∀ s ∈ S, s =
   sInf_eq_top
 #align set.sInter_eq_univ Set.sInter_eq_univ
 
+theorem subset_powerset_iff {s : Set (Set α)} {t : Set α} : ⋃₀ s ⊆ t ↔ s ⊆ 𝒫 t :=
```
Comment (YaelDillies) [inline Mathlib/Data/Set/Lattice.lean:1162]: The lemma name doesn't match the order of the statement

**Pull request closed without merging**